http://rt.cpan.org/Public/Bug/Display.html?id=41401
http://bugs.debian.org/507657
# Getting $# of an already-dereferenced-arrayref is a no-go in Perl 5.10
Index: libipc-shareable-perl/lib/IPC/Shareable.pm
===================================================================
--- libipc-shareable-perl.orig/lib/IPC/Shareable.pm	2008-12-03 10:51:27.000000000 -0600
+++ libipc-shareable-perl/lib/IPC/Shareable.pm	2008-12-03 10:51:46.000000000 -0600
@@ -452,7 +452,7 @@
     my $n    = shift;
 
     $self->{_data} = _thaw($self->{_shm}) unless $self->{_lock};
-    $#{@{$self->{_data}}} = $n - 1;
+    $#{$self->{_data}} = $n - 1;
     if ($self->{_lock} & LOCK_EX) {
         $self->{_was_changed} = 1;
     } else {

